imports/ui/Activities/TeamDiscussion/QuestionCarousel.jsx

```diff
@@ -0,0 +1,103 @@
+import ReactSwipe from 'react-swipe';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { Textfit } from 'react-textfit';
+
+class QuestionCarousel extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      prevQuestionIndex: 0,
+      startTime: new Date().getTime(),
+    };
+  }
+
+  static propTypes = {
+    pid: PropTypes.string.isRequired,
+    _id: PropTypes.string.isRequired,
+    questions: PropTypes.array.isRequired,
+    currentQuestions: PropTypes.array.isRequired,
+  };
+
+  onSlideChange = () => {
+    const endTime = new Date().getTime();
+    const { startTime } = this.state;
+
+    const { questions, _id, pid } = this.props;
+
+    const past_question = questions[this.state.prevQuestionIndex]._id;
+    const next_question = questions[this.reactSwipeEl.getPos()]._id;
+
+    //update questions
+    Meteor.call('questions.updateTimers', past_question, next_question, startTime, endTime, error => {
+      if (!error) console.log('Tracked questions successfully');
+      else console.log(error);
+    });
+
+    // keep track of this current question and when it began
+    this.setState({
+      prevQuestionIndex: this.reactSwipeEl.getPos(),
+      startTime: new Date().getTime()
+    });
+
+    Meteor.call('questions.setCurrent', _id, pid, this.reactSwipeEl.getPos(), error => {
+      if (!error) console.log('Set current question successfully');
+      else console.log(error);
+    });
+  };
+
+  getCurrentQuestion() {
+    const { pid, currentQuestions } = this.props;
+    for (var i = 0; i < currentQuestions.length; i++) {
+      if (currentQuestions[i].pid == pid) {
+        return currentQuestions[i].question_ind;
+      }
+    }
+    return 0;
+  }
+
+  render() {
+    return (
+      <div>
+        <div className="swipe-instr-top">
```

Comment:
The text 'Looks like everyone in your group has found each other!' should only appear when they start the questions early. Once the Timer starts, it should just be 'Choose questions to discuss as a group'.